test/libsolutil/StringUtils.cpp

```diff
@@ -190,7 +206,7 @@ BOOST_AUTO_TEST_CASE(test_format_number_readable_signed)
 
 	BOOST_CHECK_EQUAL(formatNumberReadable(b, true), "-0x5555...{+56 more}...5555");
 	BOOST_CHECK_EQUAL(formatNumberReadable(c, true), "-0x0BCD...{+56 more}...6789");
-	BOOST_CHECK_EQUAL(formatNumberReadable(d, true), "-0x5555555555555556 * 2**192 - 1");
+	BOOST_CHECK_EQUAL(formatNumberReadable(d, true), "-0x5555555555555556 * 2**192 + 1");
```

Comment:
Does this change has to do with the bug you mentioned, as in
> `formatNumberReadable(-0xffffffff)` should be equal to `-2 ** 32 + 1` but previously it was tested to be equal to `-2 ** 32 - 1`

?

Also, it is a bit difficult to track what these hex values translate to as decimals. How are you making sure that this is indeed the right value?

Reply:
Yes, the change is related to the bug I mentioned
Much of the test values is carried from previous values that were being tested. I only changed some of the variables to negative. I also used `isoltest` to verify the values.